### ISSUE TITLE
error in doc_generator.py

### ISSUE THREAD

**Issue by tgoetheyn** [ansible-security/security_rm_builder#6]
Hi,

I tried to get this role working, but even with the example I stumble upon errors.
I grabbed the swagger file from [trendmicro ](https://automation.deepsecurity.trendmicro.com/article/20_0/api-reference/) an used the following playbook:

```
---
- hosts: localhost
  gather_facts: yes
  vars:
    rm_swagger_json: /tmp/trendmicro-swagger.json
    rm_dest: /tmp/trendmicro/deepsec
    api_object_path: /intrusionpreventionrules/search
    module_name: 'deepsec_intrusion_prevention_rules'
    module_version: 1.2.0
    resource: intrusion_prevention_rules
    collection_org: trendmicro 
    collection_name: deepsec 
    unique_key: ""
    author: "test"
  roles:
    - ansible_security.security_rm_builder.run
```

This results in the following error in the task:
**[ansible_security.security_rm_builder.scaffold_resource_module : EXECUTE the python script]**

```json
fatal: [localhost -> 127.0.0.1]: FAILED! => 
{
  "changed": true,
  "cmd": [
    "python3",
    "doc_generator.py",
    "/tmp/trendmicro-swagger.json",
    "/intrusionpreventionrules/search",
    "deepsec_intrusion_prevention_rules",
    "1.2.0",
    "intrusion_prevention_rules",
    "trendmicro",
    "deepsec",
    "",
    "test",
    "/tmp/ansible.6prxqa9abuild"
  ],
  "delta": "0:00:00.251491",
  "end": "2022-07-06 12:05:01.312433",
  "msg": "non-zero return code",
  "rc": 1,
  "start": "2022-07-06 12:05:01.060942",
  "stderr": "Traceback (most recent call last):\n  File \"doc_generator.py\", line 910, in <module>\n    main()\n  File \"doc_generator.py\", line 885, in main\n    \"$ref\", api_object, data, global_var_mgmt_dict\n  File \"doc_generator.py\", line 614, in get_api_param_properties_recursively\n    post_object[\"properties\"] = temp_post_object[\"properties\"]\nKeyError: 'properties'",
  "stderr_lines": [
    "Traceback (most recent call last):",
    "  File \"doc_generator.py\", line 910, in <module>",
    "    main()",
    "  File \"doc_generator.py\", line 885, in main",
    "    \"$ref\", api_object, data, global_var_mgmt_dict",
    "  File \"doc_generator.py\", line 614, in get_api_param_properties_recursively",
    "    post_object[\"properties\"] = temp_post_object[\"properties\"]",
    "KeyError: 'properties'"
  ],
  "stdout": "",
  "stdout_lines": []
}
```

**Comment by justjais** [comment]
@tgoetheyn please excuse me for the delayed response, as I didn't get any notification for the issue; that said I'll triage the issue and update more asap!

**Comment by justjais** [comment]
@tgoetheyn Thanks for raising the issue, and I've checked the issue was due to wrong documentation under TM readme, I've updated the doc and if you run the sample playbook with var args, the tool will run as expected and generate the required code for TM deepsec input API.
And, thanks for sharing the play example where you've passed expected input under playbook vars. I've included the respective play in the example section as well.
